client/my-sites/marketplace/constants.ts

```diff
@@ -1,26 +1,123 @@
+
+// ********** This section to be refactored out with new definition below **********
 export default interface PluginProductMappingInterface {
 	readonly 'wordpress-seo': string;
 	readonly 'wordpress-seo-premium': string;
 }
 
-// TODO: Refactor to a product indexed map of plugins
 export const PLUGIN_PRODUCT_MAP: PluginProductMappingInterface = {
-	'wordpress-seo': YOAST_SEO,
-	'wordpress-seo-premium': YOAST_SEO,
+	'wordpress-seo': YOAST_PREMIUM,
+	'wordpress-seo-premium': YOAST_PREMIUM,
 };
 
 export function getProductSlug( pluginSlug: keyof PluginProductMappingInterface ): string {
 	return PLUGIN_PRODUCT_MAP[ pluginSlug ];
 }
+// ************************************************************************************
+
+// TODO: Integrate this data structure with marketplace code
+export const productGroups: IProductGroupCollection = {
+	[ YOAST ]: {
+		products: {
+			[ YOAST_PREMIUM ]: {
+				defaultPluginSlug: 'wordpress-seo',
+				pluginsToBeInstalled: [ 'wordpress-seo-premium', 'wordpress-seo', 'yoast-test-helper' ],
+				isPurchasableProduct: true,
+			},
+			[ YOAST_FREE ]: {
+				defaultPluginSlug: 'wordpress-seo',
+				pluginsToBeInstalled: [ 'wordpress-seo' ],
+				isPurchasableProduct: false,
+			},
+		},
+	},
+};
+
+/**
+ * Provides the plugin that needs to be installed for a given product
+ *
+ * @param {string} productGroupSlug The product group the product belongs to
+ * @param {string} productSlug The product slug being setup.
+ * @returns {array[string]} An array of plugin slugs, returns null if product does slug not exist
+ */
+export function getPluginsToInstall(
+	productGroupSlug: keyof IProductGroupCollection,
+	productSlug: keyof IProductCollection
+): string[] | null {
+	const { pluginsToBeInstalled } = productGroups[ productGroupSlug ]?.products[ productSlug ] ?? {};
+	return pluginsToBeInstalled ? pluginsToBeInstalled : null;
+}
+
+/**
+ * Provides the first product found in the product group definition
+ *
+ * @param {string} productGroupSlug The product group required.\
+ * @returns {string} The product slug
+ */
+export function getDefaultProductInProductGroup(
+	productGroupSlug: keyof IProductGroupCollection
+): string | null {
+	const { products } = productGroups[ productGroupSlug ] ?? {};
+	if ( products ) {
+		const productKeys = Object.keys( products );
+		const [ product ] = productKeys;
+		return product ? product : null;
+	}
+	return null;
+}
+
+/**
+ * Provides the first plugin found in the product definitions
+ *
+ * @param {string} productGroupSlug The product group the product belongs to
+ * @param {string} productSlug The product slug being setup
+ * @returns {string} The plugin slug
+ */
+export function getDefaultPluginInProduct(
+	productGroupSlug: keyof IProductGroupCollection,
+	productSlug: keyof IProductCollection
+): string | null {
+	const { defaultPluginSlug } = productGroups[ productGroupSlug ]?.products[ productSlug ] ?? {};
+	return defaultPluginSlug ? defaultPluginSlug : null;
```

Comment:
`defaultPluginSlug` will default to an empty object if nullish. That way it will always be truthy, so the check feels unnecessary.

Reply:
I will be restructuring the result of the nullish coalescing operator. So it can be undefined.